[Python] querying articles batch 1 [Replace C# samples] (#1797)
ravendb/docs: Documentation/5.4/Raven.Documentation.Pages/client-api/session/querying/sort-query-results.python.markdown

  
    * [Order by score](../../../client-api/session/querying/sort-query-results#order-by-score)  
        * [Get resulting score](../../../client-api/session/querying/sort-query-results#get-resulting-score)
     
    * [Order by random](../../../client-api/session/querying/sort-query-results#order-by-rando)   
     
    * [Order by spatial](../../../client-api/session/querying/sort-query-results#order-by-spatial)
     
    * [Order by count (aggregation query)](../../../client-api/session/querying/sort-query-results#order-by-count-(aggregation-query))
     
    * [Order by sum (aggregation query)](../../../client-api/session/querying/sort-query-results#order-by-sum-(aggregation-query))
     
    * [Force ordering type](../../../client-api/session/querying/sort-query-results#force-ordering-type)
     
    * [Chain ordering](../../../client-api/session/querying/sort-query-results#chain-ordering)
     
    * [Custom sorters](../../../client-api/session/querying/sort-query-results#custom-sorters)
     
    * [Syntax](../../../client-api/session/querying/sort-query-results#syntax)

{NOTE/}

---

{PANEL: Order by field value}

> * Use `OrderBy` or `OrderByDescending` to order the results by the specified document-field.

Reviewer: ```suggestion
* Use `order_by` or `order_by_descending` to order the results by the specified document-field.
```

Author: done

{CODE-TABS}
{CODE-TAB:python:Query sort_1@ClientApi\Session\Querying\SortQueryResults.py /}
{CODE-TAB-BLOCK:sql:RQL}
from "Products"

> where UnitsInStock > 10

Reviewer: ```suggestion
where units_in_stock > 10
```

Author: done

order by UnitsInStock as long
{CODE-TAB-BLOCK/}
{CODE-TABS/}

{INFO: }

**Ordering Type**:

* By default, the `order_by` methods will determine the `OrderingType` from the property path expression  
  and specify that ordering type in the generated RQL that is sent to the server.  

* E.g. in the above example, ordering by `x => x.units_in_stock` will result in `OrderingType.LONG`  
  because that property data type is an integer.

* Different ordering can be forced - see [Force ordering type](../../../client-api/session/querying/sort-query-results#force-ordering-type) below.

{INFO/}

{PANEL/}

{PANEL: Order by score}

* When querying with some filtering conditions, a basic score is calculated for each item in the results  
  by the underlying indexing engine. (Read more about Lucene scoring [here](https://lucene.apache.org/core/3_3_0/scoring.html)).

* The higher the score value the better the match.  

* Use `order_by_score` or `order_by_score_descending` to order the query results by this score.

{CODE-TABS}
{CODE-TAB:python:Query sort_4@ClientApi\Session\Querying\SortQueryResults.py /}
{CODE-TAB-BLOCK:sql:RQL}
from "Products"

> where UnitsInStock < 5 or Discontinued == true

Reviewer: ```suggestion
where units_in_stock < 5 or discontinued == true
```

Author: done

order by score()
{CODE-TAB-BLOCK/}
{CODE-TABS/}

{INFO: }

#### Get resulting score:

---

The score details can be retrieved by either:  
 
  * **Request to include explanations**:  
    You can get the score details and see how it was calculated by requesting to include explanations in the query. 
    Currently, this is only available when using Lucene as the underlying indexing engine.  
    Learn more in [Include query explanations](../../../client-api/session/querying/debugging/include-explanations).
   
  * **Get score from metadata**:  
    The score is available in the `@index-score` metadata property within each result.  
    
{INFO/}

{PANEL/}

{PANEL: Order by random}

> * Use `RandomOrdering` to randomize the order of the query results.

Reviewer: ```suggestion
* Use `RandomOrdering` to randomize the order of the query results.
```
```suggestion
* Use `random_ordering` to randomize the order of the query results.
```

Author: done

{PANEL: Force ordering type}

* By default, the `order_by` methods will determine the `OrderingType` from the property path expression  
  and specify that ordering type in the generated RQL that is sent to the server.

* A different ordering can be forced by passing the ordering type explicitly to `order_by` or `order_by_descending`.

* The following ordering types are available:

    * `OrderingType.LONG`
    * `OrderingType.DOUBLE`
    * `OrderingType.ALPHA_NUMERIC`
    * `OrderingType.STRING` (lexicographic ordering)

* When using RQL directly, if no ordering type is specified, then the server defaults to lexicographic ordering.

{NOTE: }

**Using alphanumeric ordering example**:

* When ordering mixed-character strings by the default lexicographical ordering  
  then comparison is done character by character based on the Unicode values.  
  For example, "Abc9" will come after "Abc10" since 9 is greater than 1.

* If you want the digit characters to be ordered as numbers then use alphanumeric ordering  
  where "Abc10" will result after "Abc9".

{CODE-TABS}
{CODE-TAB:python:Query sort_16@ClientApi\Session\Querying\SortQueryResults.py /}
{CODE-TAB-BLOCK:sql:RQL}
from "Products"

> order by QuantityPerUnit as alphanumeric

Author: done
